code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java

```diff
@@ -372,11 +384,6 @@ void showInRootViewGroup() {
     /** Tears down views and listeners used to display the {@link AEPMessage}. */
     void cleanup() {
         Log.trace(ServiceConstants.LOG_TAG, TAG, "Cleaning the AEPMessage.");
-        delegateFullscreenMessageDismiss();
-
-        if (messagesMonitor != null) {
```

Comment:
Can you add this back? This method is called from `MessageWebViewRunner` incase of failure and core will not display any other message unless the flag is cleared in messages monitor.

Reply:
I prefer calling `onShow(..)` listeners in viewed() and `onDismissed(...)` listeners in cleanup() if `isVisible` flag is true. 
@sbenedicadb Thoughts?